packages/transactional-adapters/transactional-adapter-kysely/src/lib/transactional-adapter-kysely.ts

```diff
@@ -82,3 +96,38 @@ const txBuilder = async <DB>(
 
     return transaction.execute();
 };
+
+class SavepointTransaction<DB> {
```

Comment:
I'm sorry, but this is overengineered to oblivion. I'm going to merge it anyway to relieve you from the responsibility that you've taken upon yourself and simplify it on my end. I also want the nested propagation feature to move forward.